src/pyconnect/core.py

```diff
@@ -26,6 +28,34 @@ def message_repr(msg: Message) -> str:
     )
 
 
+def hash_sensitive_values(
+    config: Dict[str, Any], algorithm: str = "sha256", iterations: int = 100000, log_params: bool = True
+) -> Dict[str, Any]:
+    """
+    This function takes a kakfa configuration dictionary and hashes all present sensitive values (i.e. any keys from
+    "ssl.key.password", "ssl.keystore.password", "sasl.password", "ssl.key.pem", "ssl_key" which are in the dictionary).
+    By default the hashing parameters are logged. If you do not want this behavior, set `log_params` to False.
+    Note: The original dictionary is not modified.
+
+    :param config: Kafka config dictionary.
+    :param algorithm: Hash algorithm.
+    :param iterations: Number of times to run the hashing algorithm.
+    :param log_params: Should the hashing parameters be logged?
+    :return: a dictionary in which the sensitive values are hashed.
+    """
+    SENSITIVE_KEYS = ["ssl.key.password", "ssl.keystore.password", "sasl.password", "ssl.key.pem", "ssl_key"]
+
+    config_copy = config.copy()
+    salt = os.urandom(32)
+    for key in SENSITIVE_KEYS:
+        if key in config_copy:
+            if log_params:
```

Comment:
I'd actually change the value accordingly. E.g.:
```python
            hashed_password = hashlib.pbkdf2_hmac(algorithm, config_copy[key].encode(), salt, iterations).hex()
            if log_params:
                config_copy[key] = f"$PBKDF2-HMAC-{algorithm.upper()}:{salt.hex()}:{iterations}${hashed_password}"
            else:
                config_copy[key] = hashed_password
```
That way all information is in a single log message and you don't have to skim the logs and guess which ones belong together.